crates/bevy_core_pipeline/src/bloom/mod.rs

```diff
@@ -602,59 +622,16 @@ fn prepare_bloom_textures(
     }
 }
 
-#[derive(ShaderType)]
-struct BloomUniform {
+/// The uniform struct extracted from [`BloomSettings`] attached to a [`Camera`].
+/// Will be available for use in the Bloom shader.
+#[derive(Component, ShaderType, Clone)]
+pub struct BloomUniform {
```

Comment:
Can we keep this struct private? I don't like exposing bloom implementation details.

Reply:
I'd like that too, but by doing `impl ExtractComponent` on something public like `BloomSettings` and having the `Out` type be `BloomUniform` means that `BloomUniform` too must be public (else compile error)

Reply:
You can use `#[doc(hidden]` to reduce public visibility.

Reply:
I don't love this, but I think the reduction in boilerplate is worth it, given that we can somewhat mitigate it with doc hidden (does anyone know if rust analyzer ignores hidden items when it comes to autocomplete?). I'm good with these changes.

Reply:
I added the hidden attr now.
I kept the docstring anyway since I saw that was done in other instances of hiding pub items in Bevy.